src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -170,6 +174,11 @@ public int Seek(int char0)
                 }
                 while (block.End != index)
                 {
+                    if (_processed > _maxHeaderLength)
```

Comment:
return a `ResultType { int count, bool limitReached }` object and add a `limit` into each function and the caller can decide what to do? (e.g. throw BadRequest)